### PR TITLE
Pt1: Studio Explorer in Dev Tools

### PR DESCRIPTION
**Pull request by mayakoneval**
**This PR**
Replaces GraphiQL with Apollo Studio Explorer iframed in the dev tools. This includes post messaging communication for operations in dev tools. 

**By Commit**
- e66db93: remove `graphiql-forked` and `graphiql-explorer` from our package-lock and re-install
- 59967ee: remove references to the above packages ^ (ripping out all that code)
- 01c1c29: put the initial iframe for embedded explorer in the DOM
- 9fdd8d4: set up dev tools to be able to accept a post message for when the embedded explorer has loaded, and post message the introspection schema back to the dev tools. 
- 3ffd68a: hook up communication for post messaged query requests from embedded explorer to `executeOperation`. there is a future PR where I change the name of the graphiql references throughout the relay
- 21ced8c: Previously in dev tools, when a query or mutation had an error, it just hung. This commit hooks up the error handling so that we show a gql error if there is one. TODO: I still need to iron out what we can show if there is only a network error. 
- da732da: The embedded explorer should be loaded regardless of what tab a user is on, to avoid the loading screen every time the tab is switched to explorer[. We cannot modify iframe DOM](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) (or DOM wrapping an iframe it seems) so my solution to this was to have the Explorer screen in the dom at all times, but just loaded underneath the current screen. When a user clicks Explorer, they see what we expect without the loading signal (after the first load).
- 47aa0bf: I installed a linter package to make `npm run lint` work again.
- 19a565f: Some lint (and type) fixes resulting from the above change. All of these should be preventative and non functional.
- 612f2b7: Add functionality to pass through headers form explorer. We decided not to show the headers panel in explorer, so this commit can be deleted, but I decided to keep this commit in the PR incase we wanted it in the future. Folks using AC dev tools will need to make sure their link chain forwards all headers through - which was friction enough for the Nebula team heads to decide to not include this. 
- a0d0972: add the readme explaining the above ^. Again I think this will be deleted.
- d055d0c: Subscriptions have a different `postMessage` name from explorer, so we know to keep the connection open. In this PR, we treat them the same as we do queries and mutations (after one response only, we stop listening).
- 57d2b35: The Run In Explorer button will open the explorer tab & send a `postMessage` to the embedded explorer with the operation and variables to prefill.
- 5239232: rename variables with GraphiQL in the name to Explorer.
- 5414c2f: rename `graphiqlRelay.ts` to `explorerRelay.ts`
- 17db6fc: update Explorer test to check if the embedded explorer iframe is loaded and if the header makes sense.
- ec8f1e9: Nav test checks for Explorer instead of GraphiQL
- aa31615: Add a Changelog entry that should be scrutinized. 
- d366de9: I added typing to the components, so now the tests are expecting types. Added mock props here.
- ff306f9: I had the places that I passed `fetchPolicy: queryCache` flipped between `executeOperation` for queries & mutations and the introspection query. This should make fetching from the cache work again :)

Notes: Subscriptions in dev tools weren't live previously. In this PR, I matched the previous functionality of subscriptions where there are responses populated when the play button was clicked. In a near future PR I will make live subscriptions work again. 

Question: i don't think we can use jest for iframes? It just shows the immediate DOM and not what is inside the iframe unfortunately. Happy to implement any suggestions.

**Video of chrome on studio-staging**

https://user-images.githubusercontent.com/14367451/128949249-07a8d97f-6ffd-4805-972d-74efe7b2e7fc.mov

**VIdeo of subscriptions - this is the same way subscriptions currently work in dev tools. This will be changing v soon**
https://user-images.githubusercontent.com/14367451/128949432-cf1ad124-310b-4be7-95e1-5a83872dfb69.mov

### CODE REVIEW

**Comment by mayakoneval** [comment]
@hwillson we are QA'ing this now so I might add some bug - fix commits if thats not too disturbing 🙏

**Review by hwillson — APPROVED** [review]
Really incredible work @mayakoneval - this is AWESOME! 🎉 Everything looks really great, and my very few comments are all minor and shouldn't hold up merging. One thing to call out though - we'll probably release this as part of 4.0, so would you mind changing the base branch of this PR from `main` to `release-4.0`, before merging? The `release-4.0` branch is now in place, along with a 4.0 PR to track progress here: https://github.com/apollographql/apollo-client-devtools/pull/660

Thanks again!

File: src/application/components/Explorer/explorerRelay.ts
Comment (mayakoneval) [reply src/application/components/Explorer/explorerRelay.ts:13]: I forgot to come back to this, I tried and broke everything and meant to try again. Will do 👍

**Pull request merged**
